addons/mail/models/mail_thread.py

```diff
@@ -2629,37 +2630,51 @@ def _notify_classify_recipients(self, recipient_data, model_name, msg_vals=None)
         }]
         only return groups with recipients
         """
+        recipient_lang_groups = self.env['res.partner'].sudo().read_group(
+            [('id', 'in', [recipient['id'] for recipient in recipient_data])],
+            ['lang', 'ids:array_agg(id)'],
+            ['lang']
+        )
+        partner_id_to_recipient_data = {
+            recipient['id']: recipient for recipient in recipient_data
+        }
+        recipient_langs = [group['lang'] for group in recipient_lang_groups] or [self._fallback_lang().env.context['lang']]
         # keep a local copy of msg_vals as it may be modified to include more information about groups or links
         local_msg_vals = dict(msg_vals) if msg_vals else {}
-        groups = self._notify_get_groups(msg_vals=local_msg_vals)
+        groups_by_lang = {lang: self.with_context(lang=lang)._notify_get_groups(msg_vals=local_msg_vals) for lang in recipient_langs}
         access_link = self._notify_get_action_link('view', **local_msg_vals)
 
-        if model_name:
-            view_title = _('View %s', model_name)
-        else:
-            view_title = _('View')
-
         # fill group_data with default_values if they are not complete
-        for group_name, group_func, group_data in groups:
-            group_data.setdefault('notification_group_name', group_name)
-            group_data.setdefault('notification_is_customer', False)
-            is_thread_notification = self._notify_get_recipients_thread_info(msg_vals=msg_vals)['is_thread_notification']
-            group_data.setdefault('has_button_access', is_thread_notification)
-            group_button_access = group_data.setdefault('button_access', {})
-            group_button_access.setdefault('url', access_link)
-            group_button_access.setdefault('title', view_title)
-            group_data.setdefault('actions', list())
-            group_data.setdefault('recipients', list())
+        for lang, lang_groups in groups_by_lang.items():
```

Comment:
So I suppose this is the "real" fix (a bit like what is done in _notify_get_classified_recipients_iterator in newer versions), looping on langs then each lang has recipients divided into groups (user, portal, ...) ?